src/shared/evntprov.rs

```diff
@@ -0,0 +1,286 @@
+STRUCT!{struct EVENT_FILTER_EVENT_NAME {
+    MatchAnyKeyword: ULONGLONG,
+    MatchAllKeyword: ULONGLONG,
+    Level: UCHAR,
+    FilterIn: BOOLEAN,
+    NameCount: USHORT,
+    Names: [UCHAR; ANYSIZE_ARRAY],
+}}
+pub type PEVENT_FILTER_EVENT_NAME = *mut EVENT_FILTER_EVENT_NAME;
+ENUM!{enum EVENT_INFO_CLASS {
+    EventProviderBinaryTrackInfo,
+    EventProviderSetEnableAllKeywords,
+    EventProviderSetTraits,
+    EventProviderUseDescriptorType,
+    MaxEventInfo,
+}}
+FN!{stdcall PENABLECALLBACK(
+    SourceId: LPCGUID,
+    IsEnabled: ULONG,
+    Level: UCHAR,
+    MatchAnyKeyword: ULONGLONG,
+    MatchAllKeyword: ULONGLONG,
+    FilterData: PEVENT_FILTER_DESCRIPTOR,
+    CallbackContext: PVOID,
+) -> ()}
+extern "system" {
+    pub fn EventRegister(
+        ProviderId: LPCGUID,
+        EnableCallback: PENABLECALLBACK,
+        CallbackContext: PVOID,
+        RegHandle: PREGHANDLE,
+    ) -> ULONG;
+    pub fn EventUnregister(
+        RegHandle: REGHANDLE,
+    ) -> ULONG;
+    pub fn EventSetInformation(
+        RegHandle: REGHANDLE,
+        InformationClass: EVENT_INFO_CLASS,
+        EventInformation: PVOID,
+        InformationLength: ULONG,
+    ) -> ULONG;
+    pub fn EventEnabled(
+        RegHandle: REGHANDLE,
+        EventDescriptor: PCEVENT_DESCRIPTOR,
+    ) -> BOOLEAN;
+    pub fn EventProviderEnabled(
+        RegHandle: REGHANDLE,
+        Level: UCHAR,
+        Keyword: ULONGLONG,
+    ) -> BOOLEAN;
+    pub fn EventWrite(
+        RegHandle: REGHANDLE,
+        EventDescriptor: PCEVENT_DESCRIPTOR,
+        UserDataCount: ULONG,
+        UserData: PEVENT_DATA_DESCRIPTOR,
+    ) -> ULONG;
+    pub fn EventWriteTransfer(
+        RegHandle: REGHANDLE,
+        EventDescriptor: PCEVENT_DESCRIPTOR,
+        ActivityId: LPCGUID,
+        RelatedActivityId: LPCGUID,
+        UserDataCount: ULONG,
+        UserData: PEVENT_DATA_DESCRIPTOR,
+    ) -> ULONG;
+    pub fn EventWriteEx(
+        RegHandle: REGHANDLE,
+        EventDescriptor: PCEVENT_DESCRIPTOR,
+        Filter: ULONG64,
+        Flags: ULONG,
+        ActivityId: LPCGUID,
+        RelatedActivityId: LPCGUID,
+        UserDataCount: ULONG,
+        UserData: PEVENT_DATA_DESCRIPTOR,
+    ) -> ULONG;
+    pub fn EventWriteString(
+        RegHandle: REGHANDLE,
+        Level: UCHAR,
+        Keyword: ULONGLONG,
+        EventString: PCWSTR,
+    ) -> ULONG;
+    pub fn EventActivityIdControl(
+        ControlCode: ULONG,
+        ActivityId: LPGUID,
+    ) -> ULONG;
+}
+#[inline(always)]
+pub unsafe fn EventDataDescCreate(
+    EventDataDescriptor: PEVENT_DATA_DESCRIPTOR, DataPtr: *const VOID, DataSize: ULONG
+) {
+    (*EventDataDescriptor).Ptr = DataPtr as ULONGLONG;
+    (*EventDataDescriptor).Size = DataSize;
+    *(*EventDataDescriptor).u.Reserved_mut() = 0;
+}
+#[inline(always)]
+pub unsafe fn EventDescCreate(
+    EventDescriptor: PEVENT_DESCRIPTOR, Id: USHORT, Version: UCHAR, Channel: UCHAR, Level: UCHAR,
```

Comment:
I didn't realize that style applied to function definitions. Thought it was only for the declarations. I'll go through all the files and validate that I don't have the same issue elsewhere.